lib/elixir/lib/module/checker.ex

```diff
@@ -0,0 +1,135 @@
+defmodule Module.Checker do
+  def verify(module_map, _binary) do
+    check_definitions(Enum.reverse(module_map.definitions), module_map)
```

Comment:
I am not sure why we have to reverse here. Maybe we should reverse when we build the module map instead?

Reply:
Do we have to reverse? I think it is fine without reversing.